engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentStatusMapper.scala

```diff
@@ -0,0 +1,55 @@
+package pl.touk.nussknacker.k8s.manager
+
+import io.circe.Json
+import pl.touk.nussknacker.engine.api.{CirceUtil, ProcessVersion}
+import pl.touk.nussknacker.engine.api.deployment.{ProcessState, ProcessStateDefinitionManager, StateStatus}
+import pl.touk.nussknacker.engine.api.deployment.simple.SimpleStateStatus
+import pl.touk.nussknacker.k8s.manager.K8sDeploymentManager.scenarioVersionAnnotation
+import skuber.apps.v1.Deployment
+
+//Based on https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#deployment-status
+class K8sDeploymentStatusMapper(definitionManager: ProcessStateDefinitionManager) {
+
+  private val availableCondition = "Available"
+
+  private val progressingCondition = "Progressing"
+
+  //https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
+  //"For some conditions, True represents normal operation, and for some conditions, False represents normal operation."...
+  //in our case Availability and Progressing have "positive polarity" as described in link above...
+  private val conditionStatusToCheck = "True"
+
+  private[manager] def findStatusForDeployments(deployments: List[Deployment]): Option[ProcessState] = {
+    def status(deployment: Deployment): ProcessState = {
+      val version = deployment.metadata.annotations.get(scenarioVersionAnnotation)
+        .flatMap(CirceUtil.decodeJson[ProcessVersion](_).toOption)
+      val (status, attrs, errors) = deployment.status match {
```

Comment:
wow I haven't known that we have `errors` next to `status`. pretty weird - interesting how they are presented for success statuses. I've added `EmbeddedStateStatus.DetailedFailedStateStatus` in some change - wondering if it was needed. Am i right that `attributes` are always `None`?

Reply:
Yes, attributes are always none. As for errors - I also often forget about them - they are always presented as popover I think, see ProcessStateIcon.tsx#imageWithPopover